src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalesceAssignmentTests.cs

```diff
@@ -186,5 +187,925 @@ void M(int? a)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 19));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ImplicitObjectCreationExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.True(typeInfo.Type.IsErrorType());
+            Assert.True(typeInfo.ConvertedType.IsErrorType());
+
+            var symbolInfo = model.GetSymbolInfo(node);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Empty(symbolInfo.CandidateSymbols);
+
+            var assignmentNode = descendantNodes.OfType<AssignmentExpressionSyntax>().Single();
+            var assignmentTypeInfo = model.GetTypeInfo(assignmentNode);
+            Assert.True(assignmentTypeInfo.Type.IsErrorType());
+            Assert.True(assignmentTypeInfo.ConvertedType.IsErrorType());
+        }
+
+        [Theory]
+        [InlineData("=")]
+        [InlineData("??=")]
+        public void ErrorRecovery_ImplicitObjectCreation_UnconstrainedGenericType(string assignmentOperator)
+        {
+            var source = $$"""
+                class C
+                {
+                    void M<T>() where T : new()
+                    {
+                        T t = default;
+                        t {{assignmentOperator}} new(a);
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,17): error CS0103: The name 'a' does not exist in the current context
+                //         t = new(a);
+                // (6,19): error CS0103: The name 'a' does not exist in the current context
+                //         t ??= new(a);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 16 + assignmentOperator.Length));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ImplicitObjectCreationExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+            Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+            var symbolInfo = model.GetSymbolInfo(node);
+            Assert.Null(symbolInfo.Symbol);
+            // INCONSISTENCY: Having type itself as a symbol info candidate for a constructor invocation node is odd.
+            // If there were no errors symbol info would be completely empty (no symbol & no candidates).
+            // Given that we know type of the expression, shouldn't symbol info be identical for normal and error cases?
```

Comment:
I don't really understand this comment. What do you mean that if there were no errors there'd be no symbol info? Is there an example somewhere here to better illustate the inconsistency?

Reply:
> What do you mean that if there were no errors there'd be no symbol info?

If there were no errors on the right side of the expression returned symbol info would be just a `default`:
![devenv_eJ2aWsH9u5](https://github.com/user-attachments/assets/1ad493ce-cf4a-439f-9be1-8ffc9047db7c)

> Is there an example somewhere here to better illustate the inconsistency?

`new T()` is syntactically a constructor. Therefore I would expect `GetSymbolInfo` return a constructor metho symbol or nothing when it cannot be determined. Here `GetSymbolInfo` returns nothing for a happy path, but for error recovery starts to (unexpectedly to me) return _type symbol_ T. This is inconsistent to what `GetSymbolInfo` usually returns for syntactic constructor calls, isn't it?

Reply:
Is there a test case to illustrate the whole behavior here?

Reply:
Looked for `new T()` text in the codebase and it seems like there aren't any tests which specificly verify public semantic model API output for it. Gonna add a few

Reply:
Added equivalent cases without error to demonstrate, that symbol info in such scenarios is always empty